### PR TITLE
SP-3765: Oracles. Local debug

### PR DESCRIPTION
**Pull request by marchuk-vlad**
Before launch make sure that your local port 8545 is free.
```
git clone https://github.com/Super-Protocol/solutions.git
cd solutions/Blockchain/sp-oracle/smart-contract/
npm run compile
cd ../script/
docker compose up build
docker compose up blockchain oracle
```

`The iteration of oracle loop has been ended` log means that everything is OK and you can check data in blockchain. 
Please open file `solutions/Blockchain/sp-oracle/smart-contract/test-app-commands.txt` and invoke commands that you find there in `smart-contract` folder.

### CODE REVIEW

File: Blockchain/sp-oracle/script/src/common/intrefaces.ts [hunk 1]
```diff
@@ -27,6 +27,7 @@ export interface IPubService {
 }
 
 export interface IQuoteProvider {
+  initialiaze(): Promise<void>;
```
Comment (Idris0v) [inline Blockchain/sp-oracle/script/src/common/intrefaces.ts:30]: initialiaze -> initialize

File: Blockchain/sp-oracle/script/src/providers/quote/getQuoteProvider.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+};
+
+export const getQuoteProvider = (): IQuoteProvider => {
+  const isMockQuote = process.env.NODE_ENV === 'test';
+
+  if (isMockQuote) {
+    printDebugModeWarning();
+
+    const printDebugModeWarningInterval = 5 * 60 * 1000; //5 min
+    setInterval(() => printDebugModeWarning(), printDebugModeWarningInterval);
```
Comment (Idris0v) [inline Blockchain/sp-oracle/script/src/providers/quote/getQuoteProvider.ts:20]: that is, log 5 times in the beginning, then repeat every 5 minutes? Maybe once is enough...

File: Blockchain/sp-oracle/script/tsconfig.json [hunk 3]
```diff
@@ -1,17 +1,19 @@
...
+    "outDir": "run/dist",
+    "allowJs": true,
+    "strict": true,
+    "declaration": true,
+    "noImplicitAny": true,
+    "esModuleInterop": true,
+    "resolveJsonModule": true,
+    "inlineSourceMap": true,
+    "incremental": true,
+    "tsBuildInfoFile": "run/dis/tsconfig.build.tsbuildinfo"
```
Comment (Idris0v) [inline Blockchain/sp-oracle/script/tsconfig.json:16]: typo: `run/dist/tsconfig.build.tsbuildinfo`

File: Blockchain/sp-oracle/script/src/providers/quote/getQuoteProvider.ts (hunk 2 above)
Comment (marchuk-vlad) [reply Blockchain/sp-oracle/script/src/providers/quote/getQuoteProvider.ts:20]: Yep, it's not our common pattern, but at the beginning, logs from Oracle are so many, and this warning cannot be seen by the user.

File: Blockchain/sp-oracle/smart-contract/contracts/mock/VerificatorMock.sol [hunk 4]
```diff
@@ -0,0 +1,33 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+
+contract VerificatorMock {
```
[the remark this answers is not in the document]
Comment (marchuk-vlad) [reply Blockchain/sp-oracle/smart-contract/contracts/mock/VerificatorMock.sol:5]: Ok, lets do it as separete task

**Review by AgentRX — APPROVED** [review]

**Pull request merged**